src/test/java/ch/srgssr/playfff/controller/RecommendationServiceRSIOneFullLengthVideoResultTests.java

```diff
@@ -0,0 +1,282 @@
+        integrationLayerRequest.setRestTemplate(savedRestTemplate);
+    }
+
+    @Test
+    public void videoFullLengthAndNextURLforOldestEpisodeTest() throws URISyntaxException {
+        String urn = "urn:rsi:video:1";
+        List<String> expectedUrns = Arrays.asList("urn:rsi:video:2", "urn:rsi:video:3", "urn:rsi:video:4");
+        List<String> expectedStandaloneUrns = Arrays.asList("urn:rsi:video:2", "urn:rsi:video:3", "urn:rsi:video:4");
+
+        testVideoRecommendation(urn, true,true, expectedUrns, expectedStandaloneUrns);
+    }
+
+    @Test
+    public void videoFullLengthforOldestEpisodeTest() throws URISyntaxException {
+        String urn = "urn:rsi:video:1";
+        List<String> expectedUrns = Arrays.asList("urn:rsi:video:2", "urn:rsi:video:3", "urn:rsi:video:4");
+        List<String> expectedStandaloneUrns = Arrays.asList("urn:rsi:video:2", "urn:rsi:video:3", "urn:rsi:video:4");
+
+        testVideoRecommendation(urn, true,false, expectedUrns, expectedStandaloneUrns);
+    }
+
+    @Test
+    public void videoFullLengthAndNextURLforSecondEpisodeTest() throws URISyntaxException {
+        String urn = "urn:rsi:video:2";
+        List<String> expectedUrns = Arrays.asList("urn:rsi:video:3", "urn:rsi:video:4", "urn:rsi:video:1");
+        List<String> expectedStandaloneUrns = Arrays.asList("urn:rsi:video:3", "urn:rsi:video:4", "urn:rsi:video:1");
+
+        testVideoRecommendation(urn, true,true, expectedUrns, expectedStandaloneUrns);
+    }
+
+    @Test
+    public void videoFullLengthforSecondEpisodeTest() throws URISyntaxException {
+        String urn = "urn:rsi:video:2";
+        List<String> expectedUrns = Arrays.asList("urn:rsi:video:3", "urn:rsi:video:4", "urn:rsi:video:1");
+        List<String> expectedStandaloneUrns = Arrays.asList("urn:rsi:video:3", "urn:rsi:video:4", "urn:rsi:video:1");
+
+        testVideoRecommendation(urn, true,false, expectedUrns, expectedStandaloneUrns);
+    }
+
+    @Test
+    public void videoFullLengthAndNextURLforThirdEpisodeTest() throws URISyntaxException {
+        String urn = "urn:rsi:video:3";
+        List<String> expectedUrns = Arrays.asList("urn:rsi:video:4", "urn:rsi:video:2", "urn:rsi:video:1");
+        List<String> expectedStandaloneUrns = Arrays.asList("urn:rsi:video:4", "urn:rsi:video:2", "urn:rsi:video:1");
+
+        testVideoRecommendation(urn, true,true, expectedUrns, expectedStandaloneUrns);
+    }
+
+    @Test
+    public void videoFullLengthforThirdEpisodeTest() throws URISyntaxException {
+        String urn = "urn:rsi:video:3";
+        List<String> expectedUrns = Arrays.asList("urn:rsi:video:4", "urn:rsi:video:1", "urn:rsi:video:2");
+        List<String> expectedStandaloneUrns = Arrays.asList("urn:rsi:video:4", "urn:rsi:video:1", "urn:rsi:video:2");
+
+        testVideoRecommendation(urn, true,false, expectedUrns, expectedStandaloneUrns);
+    }
+
+    @Test
+    public void videoFullLengthAndNextURLforNewestEpisodeTest() throws URISyntaxException {
```

Comment:
for readability purpose instead of camel case in unit test it could be great to use snake case instead : video_full_length_and_next_URL_for_newest_episode_test
@pyby, @SebastienChauvin